src/bun.js/child_process.exports.js

```diff
@@ -1,3 +1,1712 @@
+      const err = errnoException(exitCode, syscall);
+
+      if (this.spawnfile) err.path = this.spawnfile;
+
+      err.spawnargs = ArrayPrototypeSlice.call(this.spawnargs, 1);
+      this.emit("error", err);
+    } else {
+      this.emit("exit", this.exitCode, this.signalCode);
+    }
+
+    // If any of the stdio streams have not been touched,
+    // then pull all the data through so that it can get the
+    // eof and emit a 'close' event.
+    // Do it on nextTick so that the user has one last chance
+    // to consume the output, if for example they only want to
+    // start reading the data once the process exits.
+    process.nextTick(flushStdio, this);
+
+    this.#maybeClose();
+    this.#exited = true;
+  }
+
+  #getBunSpawnIo(stdio, options) {
+    const result = [];
+    switch (stdio[0]) {
+      case "pipe":
+        result[0] = new WrappedFileSink(this.#handle.stdin);
+        break;
+      case "inherit":
+        result[0] = process.stdin;
+      default:
+        result[0] = null;
+    }
+    let i = 1;
+    for (; i < stdio.length; i++) {
+      switch (stdio[i]) {
+        case "pipe":
+          result[i] = ReadableFromWeb(this.#handle[fdToStdioName(i)], {
+            encoding: options.encoding || undefined,
+          });
+          break;
+        case "inherit":
+          result[i] = process[fdToStdioName(i)];
+          break;
+        default:
+          result[i] = null;
+      }
+    }
+    return result;
+  }
+
+  spawn(options) {
+    validateObject(options, "options");
+
+    // validateOneOf(options.serialization, "options.serialization", [
+    //   undefined,
+    //   "json",
+    //   // "advanced", // TODO
+    // ]);
+    // const serialization = options.serialization || "json";
+
+    // if (ipc !== undefined) {
+    //   // Let child process know about opened IPC channel
+    //   if (options.envPairs === undefined) options.envPairs = [];
+    //   else validateArray(options.envPairs, "options.envPairs");
+
+    //   ArrayPrototypePush.call(options.envPairs, `NODE_CHANNEL_FD=${ipcFd}`);
+    //   ArrayPrototypePush.call(
+    //     options.envPairs,
+    //     `NODE_CHANNEL_SERIALIZATION_MODE=${serialization}`
+    //   );
+    // }
+
+    validateString(options.file, "options.file");
+    this.spawnfile = options.file;
+
+    if (options.args === undefined) {
+      this.spawnargs = [];
+    } else {
+      validateArray(options.args, "options.args");
+      this.spawnargs = options.args;
+    }
+
+    const stdio = options.stdio || "pipe";
+    const bunStdio = getBunStdioOptions(stdio);
+
+    const cmd = options.args;
+    this.#handle = Bun.spawn({
+      cmd,
+      stdin: bunStdio[0],
+      stdout: bunStdio[1],
+      stderr: bunStdio[2],
+      cwd: options.cwd || undefined,
+      env: options.envPairs || undefined,
+      onExit: this.#handleOnExit.bind(this),
+    });
+
+    this.stdio = this.#getBunSpawnIo(bunStdio, options);
+    this.stdin = this.stdio[0];
```

Comment:
these might need to be getters so that the values are the same/not copied but i'm not 100% sure